src/database/models/organization_item.rs

```diff
@@ -0,0 +1,361 @@
+            .try_collect::<Vec<Organization>>()
+            .await?;
+
+            for organization in organizations {
+                cmd("SET")
+                    .arg(format!("{}:{}", ORGANIZATIONS_NAMESPACE, organization.id.0))
+                    .arg(serde_json::to_string(&organization)?)
+                    .arg("EX")
+                    .arg(DEFAULT_EXPIRY)
+                    .query_async::<_, ()>(&mut redis)
+                    .await?;
+
+                cmd("SET")
+                    .arg(format!(
+                        "{}:{}",
+                        ORGANIZATIONS_SLUGS_NAMESPACE,
+                        organization.slug.to_lowercase()
+                    ))
+                    .arg(organization.id.0)
+                    .arg("EX")
+                    .arg(DEFAULT_EXPIRY)
+                    .query_async::<_, ()>(&mut redis)
+                    .await?;
+                found_organizations.push(organization);
+            }
+        }
+
+        Ok(found_organizations)
+    }
+
+    // Gets organization associated with a project ID, if it exists and there is one
+    pub async fn get_associated_organization_project_id<'a, 'b, E>(
+        project_id: ProjectId,
+        exec: E,
+    ) -> Result<Option<Self>, super::DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        let result = sqlx::query!(
+            "
+            SELECT o.id, o.name, o.slug, o.team_id, o.description, o.default_project_permissions, o.urls, o.icon_url, o.color
+            FROM organizations o
+            LEFT JOIN mods m ON m.organization_id = o.id
+            WHERE m.id = $1
+            GROUP BY o.id;
+            ",
+            project_id as ProjectId,
+        )
+        .fetch_optional(exec)
+        .await?;
+
+        if let Some(result) = result {
+            Ok(Some(Organization {
+                id: OrganizationId(result.id),
+                name: result.name,
+                slug: result.slug,
+                team_id: TeamId(result.team_id),
+                description: result.description,
+                default_project_permissions: ProjectPermissions::from_bits(
+                    result.default_project_permissions as u64,
+                )
+                .unwrap_or_default(),
+                urls: serde_json::from_str(&result.urls)?,
+                icon_url: result.icon_url,
+                color: result.color.map(|x| x as u32),
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+
+    pub async fn remove(
+        id: OrganizationId,
+        transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+        redis: &deadpool_redis::Pool,
+    ) -> Result<Option<()>, super::DatabaseError> {
+        let project = Self::get_id(id, &mut *transaction, redis).await?;
+
+        if let Some(organization) = project {
+            Organization::clear_cache(id, Some(organization.slug), redis).await?;
+
+            sqlx::query!(
```

Comment:
make sure to remove associated projects here. also if you ever link data to something else in the database it should be removed in the delete function (just for future ref)